transport/tlscommon/tlscommon_test.go

```diff
@@ -104,3 +111,56 @@ func writeTestFile(t *testing.T, content string) string {
 	require.NoError(t, err)
 	return f.Name()
 }
+
+const (
+	blockTypePKCS1 int = iota
+	blockTypePKCS8
+	blockTypePKCS1Encrypted
+	blockTypePKCS8Encrypted
+)
+
+// Setup key+cert pair for the tests
+func makeKeyCertPair(t *testing.T, blockType int, password string) (string, string) {
+	key, err := rsa.GenerateKey(rand.Reader, 2048)
+	require.NoError(t, err)
+	var block *pem.Block
+	switch blockType {
+	case blockTypePKCS1:
+		block = &pem.Block{
+			Type:  "RSA PRIVATE KEY",
+			Bytes: x509.MarshalPKCS1PrivateKey(key),
+		}
+	case blockTypePKCS8:
+		b, err := x509.MarshalPKCS8PrivateKey(key)
+		require.NoError(t, err)
+		block = &pem.Block{
+			Type:  "PRIVATE KEY",
+			Bytes: b,
+		}
+	case blockTypePKCS1Encrypted:
+		var err error
+		block, err = x509.EncryptPEMBlock(rand.Reader, "RSA PRIVATE KEY", x509.MarshalPKCS1PrivateKey(key), []byte(password), x509.PEMCipherAES256) //nolint:staticcheck // we need to support encrypted private keys
+		require.NoError(t, err)
+	case blockTypePKCS8Encrypted:
+		//TODO: this uses an elastic implementation of pkcs8 as the stdlib does not support password protected pkcs8
```

Comment:
Why a `TODO` comment? Shouldn't it be just the comment explaining why we use our custom implementation?
```suggestion
		// This uses an elastic implementation of pkcs8 as the stdlib does not support password protected pkcs8
```